### PR TITLE
[ShapeOpt] automatically add conditions in face_angle response

### PR DESCRIPTION
**Pull request by armingeiser**
**Description**
The face_angle response can be applied to a subset of the design surface only. 
With this change, it is possible to define that subset by nodes only, conditions (that are internally needed by the response) are automatically added from the design surface, if all condition nodes are also contained in the subset of nodes given. 

**Changelog**
- Allow definition of "only" model_-part in face_angle response by nodes only

### CODE REVIEW

**Comment by armingeiser** [comment]
I think there exists a utility for extracting all conditions that are contained in a set of nodes, but i could not find it...

File: applications/ShapeOptimizationApplication/python_scripts/response_functions/face_angle.py [hunk 1]
```diff
@@ -112,7 +132,7 @@ def GetValue(self):
 
     def GetNodalGradient(self, variable):
         if variable != KM.SHAPE_SENSITIVITY:
-            raise RuntimeError("GetNodalGradient: No gradient for {}!".format(variable.Name))
+            raise RuntimeError(f"GetNodalGradient: No gradient for {variable.Name}!")
```
Comment (philbucher) [inline applications/ShapeOptimizationApplication/python_scripts/response_functions/face_angle.py]: careful. f-strings are python 3.6!

**Review by philbucher — COMMENTED** [review]
@loumalouomega do you know if we have such a utility?

File: applications/ShapeOptimizationApplication/python_scripts/response_functions/face_angle.py (hunk 1 above)
Comment (loumalouomega) [reply applications/ShapeOptimizationApplication/python_scripts/response_functions/face_angle.py]: Why I have been mentioned?

Comment (armingeiser) [reply applications/ShapeOptimizationApplication/python_scripts/response_functions/face_angle.py]: - about f-strings: :+1: will switch to the old style
- @loumalouomega: You beeing mentioned was not related to the f-strings. Do you know if there is a utility, that assigns those conditions from modelpart A to modelpart B only if all nodes of the condition are included in modelpart B?

**Comment by loumalouomega** [comment]
In the auxiliar model part utilities, but I don't remember if it is merged
yet

**Comment by armingeiser** [comment]
Completely forgot about this MR:

> I think there exists a utility for extracting all conditions that are contained in a set of nodes, but i could not find it...

I double checked, such a utlity is not yet available.
So this is ready from my side.

**Review by philbucher — APPROVED** [review]

**Pull request merged**